{
  "url": "https://api.github.com/repos/facebook/react/issues/14838",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/14838/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/14838/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/14838/events",
  "html_url": "https://github.com/facebook/react/issues/14838",
  "id": 409573758,
  "node_id": "MDU6SXNzdWU0MDk1NzM3NTg=",
  "number": 14838,
  "title": "TypeError: func.apply is not a function / Uncaught TypeError: destroy is not a function",
  "user": {
    "login": "killpowa",
    "id": 10165264,
    "node_id": "MDQ6VXNlcjEwMTY1MjY0",
    "avatar_url": "https://avatars3.githubusercontent.com/u/10165264?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/killpowa",
    "html_url": "https://github.com/killpowa",
    "followers_url": "https://api.github.com/users/killpowa/followers",
    "following_url": "https://api.github.com/users/killpowa/following{/other_user}",
    "gists_url": "https://api.github.com/users/killpowa/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/killpowa/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/killpowa/subscriptions",
    "organizations_url": "https://api.github.com/users/killpowa/orgs",
    "repos_url": "https://api.github.com/users/killpowa/repos",
    "events_url": "https://api.github.com/users/killpowa/events{/privacy}",
    "received_events_url": "https://api.github.com/users/killpowa/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 8,
  "created_at": "2019-02-13T00:59:44Z",
  "updated_at": "2019-09-30T13:26:11Z",
  "closed_at": "2019-02-13T01:34:41Z",
  "author_association": "NONE",
  "body": "\r\n**Do you want to request a *feature* or report a *bug*?**\r\n\r\nA bug\r\n\r\n**What is the current behavior?**\r\n\r\nDoing certain dispatches with redux or other random things cause react to crash. It's working perfectly on react 16.8.0-alpha0 but as soon as I update to alpha 1 it crashes.\r\n\r\n```\r\nconsole.js:59 01:27:08.503 › TypeError: func.apply is not a function\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at safelyCallDestroy (http://localhost:1212/dist/renderer.dev.js:267167:5)\r\n    at commitUnmount (http://localhost:1212/dist/renderer.dev.js:267504:17)\r\n    at commitNestedUnmounts (http://localhost:1212/dist/renderer.dev.js:267548:5)\r\n    at unmountHostComponents (http://localhost:1212/dist/renderer.dev.js:267809:7)\r\n    at commitDeletion (http://localhost:1212/dist/renderer.dev.js:267861:5)\r\n    at commitAllHostEffects (http://localhost:1212/dist/renderer.dev.js:268602:11)\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at commitRoot (http://localhost:1212/dist/renderer.dev.js:268826:7)\r\n    at completeRoot (http://localhost:1212/dist/renderer.dev.js:270318:3)\r\n    at performWorkOnRoot (http://localhost:1212/dist/renderer.dev.js:270247:9)\r\n    at performWork (http://localhost:1212/dist/renderer.dev.js:270155:7)\r\nconsoleLog @ console.js:59\r\ntransport @ console.js:35\r\nlog @ log.js:25\r\ncomponentDidCatch @ Root.js:20\r\ncallback @ react-dom.development.js:17617\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\ninteractiveUpdates$1 @ react-dom.development.js:19964\r\ninteractiveUpdates @ react-dom.development.js:2169\r\ndispatchInteractiveEvent @ react-dom.development.js:4878\r\n```\r\n\r\n```\r\n01:52:09.074 › TypeError: destroy is not a function\r\n    at commitHookEffectList (http://localhost:1212/dist/renderer.dev.js:267241:11)\r\n    at commitPassiveHookEffects (http://localhost:1212/dist/renderer.dev.js:267270:3)\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at commitPassiveEffects (http://localhost:1212/dist/renderer.dev.js:268691:9)\r\n    at wrapped (http://localhost:1212/dist/renderer.dev.js:298178:34)\r\n    at flushFirstCallback (http://localhost:1212/dist/renderer.dev.js:298513:28)\r\n    at flushWork (http://localhost:1212/dist/renderer.dev.js:298625:11)\r\n    at MessagePort.module.exports../node_modules/scheduler/cjs/scheduler.development.js.channel.port1.onmessage (http://localhost:1212/dist/renderer.dev.js:298986:9)\r\nconsoleLog @ console.js:59\r\ntransport @ console.js:35\r\nlog @ log.js:25\r\ncomponentDidCatch @ Root.js:20\r\ncallback @ react-dom.development.js:17617\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\ncommitPassiveEffects @ react-dom.development.js:18278\r\nwrapped @ scheduler-tracing.development.js:207\r\nflushFirstCallback @ scheduler.development.js:107\r\nflushWork @ scheduler.development.js:219\r\nmodule.exports../node_modules/scheduler/cjs/scheduler.development.js.channel.port1.onmessage @ scheduler.development.js:580\r\n```\r\n\r\n\r\n\r\nThe problem looks like is caused here (?):\r\n\r\n```\r\n      function callCallback() {\r\n        // We immediately remove the callback from event listeners so that\r\n        // nested `invokeGuardedCallback` calls do not clash. Otherwise, a\r\n        // nested call would trigger the fake event handlers of any call higher\r\n        // in the stack.\r\n        fakeNode.removeEventListener(evtType, callCallback, false);\r\n\r\n        // We check for window.hasOwnProperty('event') to prevent the\r\n        // window.event assignment in both IE <= 10 as they throw an error\r\n        // \"Member not found\" in strict mode, and in Firefox which does not\r\n        // support window.event.\r\n        if (typeof window.event !== 'undefined' && window.hasOwnProperty('event')) {\r\n          window.event = windowEvent;\r\n        }\r\n\r\n        func.apply(context, funcArgs); <---------------------\r\n        didError = false;\r\n      }\r\n...\r\n```\r\n\r\n\r\n\r\n**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**\r\n\r\nWell, I don't really know how to reproduce but you can find my app code [here](https://github.com/gorilla-devs/GDLauncher/tree/94c610d904e6e595d4b48d1daab1acd31c6550c3), here are the important files:\r\n[configureStore.dev.js](https://github.com/gorilla-devs/GDLauncher/blob/94c610d904e6e595d4b48d1daab1acd31c6550c3/app/store/configureStore.dev.js) -> Has all configs of redux\r\n[this](https://github.com/gorilla-devs/GDLauncher/blob/94c610d904e6e595d4b48d1daab1acd31c6550c3/app/actions/auth.js#L64) and [this](https://github.com/gorilla-devs/GDLauncher/blob/94c610d904e6e595d4b48d1daab1acd31c6550c3/app/components/Common/Modal/Modal.js#L42) are just random places where it happens.\r\nHere are some more logs that might be useful:\r\n```\r\nUncaught TypeError: func.apply is not a function\r\n    at HTMLUnknownElement.callCallback (react-dom.development.js:149)\r\n    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)\r\n    at invokeGuardedCallback (react-dom.development.js:256)\r\n    at safelyCallDestroy (react-dom.development.js:16733)\r\n    at commitUnmount (react-dom.development.js:17070)\r\n    at unmountHostComponents (react-dom.development.js:17396)\r\n    at commitDeletion (react-dom.development.js:17427)\r\n    at commitAllHostEffects (react-dom.development.js:18168)\r\n    at HTMLUnknownElement.callCallback (react-dom.development.js:149)\r\n    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)\r\n    at invokeGuardedCallback (react-dom.development.js:256)\r\n    at commitRoot (react-dom.development.js:18392)\r\n    at completeRoot (react-dom.development.js:19884)\r\n    at performWorkOnRoot (react-dom.development.js:19813)\r\n    at performWork (react-dom.development.js:19721)\r\n    at performSyncWork (react-dom.development.js:19695)\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\nsafelyCallDestroy @ react-dom.development.js:16733\r\ncommitUnmount @ react-dom.development.js:17070\r\nunmountHostComponents @ react-dom.development.js:17396\r\ncommitDeletion @ react-dom.development.js:17427\r\ncommitAllHostEffects @ react-dom.development.js:18168\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18392\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\nreact-dom.development.js:149 Uncaught TypeError: func.apply is not a function\r\n    at HTMLUnknownElement.callCallback (react-dom.development.js:149)\r\n    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)\r\n    at invokeGuardedCallback (react-dom.development.js:256)\r\n    at safelyCallDestroy (react-dom.development.js:16733)\r\n    at commitUnmount (react-dom.development.js:17070)\r\n    at commitNestedUnmounts (react-dom.development.js:17114)\r\n    at unmountHostComponents (react-dom.development.js:17375)\r\n    at commitDeletion (react-dom.development.js:17427)\r\n    at commitAllHostEffects (react-dom.development.js:18168)\r\n    at HTMLUnknownElement.callCallback (react-dom.development.js:149)\r\n    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)\r\n    at invokeGuardedCallback (react-dom.development.js:256)\r\n    at commitRoot (react-dom.development.js:18392)\r\n    at completeRoot (react-dom.development.js:19884)\r\n    at performWorkOnRoot (react-dom.development.js:19813)\r\n    at performWork (react-dom.development.js:19721)\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\nsafelyCallDestroy @ react-dom.development.js:16733\r\ncommitUnmount @ react-dom.development.js:17070\r\ncommitNestedUnmounts @ react-dom.development.js:17114\r\nunmountHostComponents @ react-dom.development.js:17375\r\ncommitDeletion @ react-dom.development.js:17427\r\ncommitAllHostEffects @ react-dom.development.js:18168\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18392\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\nreact-dom.development.js:16647 The above error occurred in one of your React components:\r\n    in Unknown\r\n    in Suspense\r\n    in Unknown (created by Route)\r\n    in Route (created by RouteDef)\r\n    in StrictMode (created by App)\r\n    in App (created by RouteDef)\r\n    in RouteDef (created by Context.Consumer)\r\n    in Connect(RouteDef) (created by Root)\r\n    in Router (created by ConnectedRouter)\r\n    in ConnectedRouter (created by Context.Consumer)\r\n    in ConnectedRouterWithContext (created by Context.Consumer)\r\n    in Connect(ConnectedRouterWithContext) (created by Root)\r\n    in div (created by Root)\r\n    in Provider (created by Root)\r\n    in Root\r\n    in AppContainer\r\n\r\nReact will try to recreate this component tree from scratch using the error boundary you provided, Root.\r\nlogCapturedError @ react-dom.development.js:16647\r\nlogError @ react-dom.development.js:16683\r\ncallback @ react-dom.development.js:17616\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\nconsole.js:59 01:56:16.048 › TypeError: func.apply is not a function\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at safelyCallDestroy (http://localhost:1212/dist/renderer.dev.js:267167:5)\r\n    at commitUnmount (http://localhost:1212/dist/renderer.dev.js:267504:17)\r\n    at unmountHostComponents (http://localhost:1212/dist/renderer.dev.js:267830:7)\r\n    at commitDeletion (http://localhost:1212/dist/renderer.dev.js:267861:5)\r\n    at commitAllHostEffects (http://localhost:1212/dist/renderer.dev.js:268602:11)\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at commitRoot (http://localhost:1212/dist/renderer.dev.js:268826:7)\r\n    at completeRoot (http://localhost:1212/dist/renderer.dev.js:270318:3)\r\n    at performWorkOnRoot (http://localhost:1212/dist/renderer.dev.js:270247:9)\r\n    at performWork (http://localhost:1212/dist/renderer.dev.js:270155:7)\r\n    at performSyncWork (http://localhost:1212/dist/renderer.dev.js:270129:3)\r\nconsoleLog @ console.js:59\r\ntransport @ console.js:35\r\nlog @ log.js:25\r\ncomponentDidCatch @ Root.js:20\r\ncallback @ react-dom.development.js:17617\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\nreact-dom.development.js:16647 The above error occurred in the <SideBar> component:\r\n    in SideBar (created by Context.Consumer)\r\n    in Connect(SideBar) (created by RouteDef)\r\n    in div (created by RouteDef)\r\n    in div (created by RouteDef)\r\n    in StrictMode (created by App)\r\n    in App (created by RouteDef)\r\n    in RouteDef (created by Context.Consumer)\r\n    in Connect(RouteDef) (created by Root)\r\n    in Router (created by ConnectedRouter)\r\n    in ConnectedRouter (created by Context.Consumer)\r\n    in ConnectedRouterWithContext (created by Context.Consumer)\r\n    in Connect(ConnectedRouterWithContext) (created by Root)\r\n    in div (created by Root)\r\n    in Provider (created by Root)\r\n    in Root\r\n    in AppContainer\r\n\r\nReact will try to recreate this component tree from scratch using the error boundary you provided, Root.\r\nlogCapturedError @ react-dom.development.js:16647\r\nlogError @ react-dom.development.js:16683\r\ncallback @ react-dom.development.js:17616\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\nconsole.js:59 01:56:16.056 › TypeError: func.apply is not a function\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at safelyCallDestroy (http://localhost:1212/dist/renderer.dev.js:267167:5)\r\n    at commitUnmount (http://localhost:1212/dist/renderer.dev.js:267504:17)\r\n    at commitNestedUnmounts (http://localhost:1212/dist/renderer.dev.js:267548:5)\r\n    at unmountHostComponents (http://localhost:1212/dist/renderer.dev.js:267809:7)\r\n    at commitDeletion (http://localhost:1212/dist/renderer.dev.js:267861:5)\r\n    at commitAllHostEffects (http://localhost:1212/dist/renderer.dev.js:268602:11)\r\n    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:250583:14)\r\n    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:250633:16)\r\n    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:250690:31)\r\n    at commitRoot (http://localhost:1212/dist/renderer.dev.js:268826:7)\r\n    at completeRoot (http://localhost:1212/dist/renderer.dev.js:270318:3)\r\n    at performWorkOnRoot (http://localhost:1212/dist/renderer.dev.js:270247:9)\r\n    at performWork (http://localhost:1212/dist/renderer.dev.js:270155:7)\r\nconsoleLog @ console.js:59\r\ntransport @ console.js:35\r\nlog @ log.js:25\r\ncomponentDidCatch @ Root.js:20\r\ncallback @ react-dom.development.js:17617\r\ncallCallback @ react-dom.development.js:15983\r\ncommitUpdateEffects @ react-dom.development.js:16022\r\ncommitUpdateQueue @ react-dom.development.js:16010\r\ncommitLifeCycles @ react-dom.development.js:16894\r\ncommitAllLifeCycles @ react-dom.development.js:18219\r\ncallCallback @ react-dom.development.js:149\r\ninvokeGuardedCallbackDev @ react-dom.development.js:199\r\ninvokeGuardedCallback @ react-dom.development.js:256\r\ncommitRoot @ react-dom.development.js:18427\r\ncompleteRoot @ react-dom.development.js:19884\r\nperformWorkOnRoot @ react-dom.development.js:19813\r\nperformWork @ react-dom.development.js:19721\r\nperformSyncWork @ react-dom.development.js:19695\r\nrequestWork @ react-dom.development.js:19564\r\nscheduleWork @ react-dom.development.js:19373\r\nenqueueSetState @ react-dom.development.js:11051\r\nmodule.exports../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:335\r\n(anonymous) @ Provider.js:53\r\ndispatch @ redux.js:214\r\ne @ VM115:1\r\n(anonymous) @ middleware.js:22\r\n(anonymous) @ redux-logger.js:1\r\n(anonymous) @ index.js:11\r\ndispatch @ VM115:1\r\nonLocationChanged @ ConnectedRouter.js:139\r\nhandleLocationChange @ ConnectedRouter.js:84\r\nlistener @ createTransitionManager.js:46\r\n(anonymous) @ createTransitionManager.js:65\r\nnotifyListeners @ createTransitionManager.js:64\r\nsetState @ createHashHistory.js:86\r\n(anonymous) @ createHashHistory.js:122\r\nconfirmTransitionTo @ createTransitionManager.js:36\r\nhandlePop @ createHashHistory.js:120\r\nhandleHashChange @ createHashHistory.js:109\r\n```\r\n\r\nI tried giving you every piece of information I could, if you need any more info I'll be more than happy to help\r\n\r\n\r\n**What is the expected behavior?**\r\n\r\nWell, it shouldn't crash I guess\r\n\r\n**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**\r\n\r\nIt's working perfectly on react 16.8.0-alpha 0 but as soon as I update to alpha 1 or later it crashes.\r\n",
  "closed_by": {
    "login": "aweary",
    "id": 6886061,
    "node_id": "MDQ6VXNlcjY4ODYwNjE=",
    "avatar_url": "https://avatars2.githubusercontent.com/u/6886061?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/aweary",
    "html_url": "https://github.com/aweary",
    "followers_url": "https://api.github.com/users/aweary/followers",
    "following_url": "https://api.github.com/users/aweary/following{/other_user}",
    "gists_url": "https://api.github.com/users/aweary/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/aweary/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/aweary/subscriptions",
    "organizations_url": "https://api.github.com/users/aweary/orgs",
    "repos_url": "https://api.github.com/users/aweary/repos",
    "events_url": "https://api.github.com/users/aweary/events{/privacy}",
    "received_events_url": "https://api.github.com/users/aweary/received_events",
    "type": "User",
    "site_admin": false
  }
}
